x/qgb/keeper/keeper_data_commitment_test.go

```diff
@@ -186,3 +188,48 @@ func TestCheckingLatestAttestationNonceInDataCommitments(t *testing.T) {
 		})
 	}
 }
+
+func TestCheckingEarliestAvailableAttestationNonceInDataCommitments(t *testing.T) {
+	input := testutil.CreateTestEnvWithoutQGBKeysInit(t)
+	k := input.QgbKeeper
+
+	// init the latest attestation nonce
+	input.QgbKeeper.SetLatestAttestationNonce(input.Context, qgb.InitialLatestAttestationNonce)
+
+	tests := []struct {
+		name          string
+		requestFunc   func() error
+		expectedError error
+	}{
+		{
+			name: "check earliest available attestation nonce before getting current data commitment",
+			requestFunc: func() error {
+				_, err := k.NextDataCommitment(input.Context)
+				return err
+			},
+			expectedError: types.ErrEarliestAvailableNonceStillNotInitialized,
+		},
+		{
+			name: "check earliest available attestation nonce before getting data commitment for height",
+			requestFunc: func() error {
+				_, err := k.GetDataCommitmentForHeight(input.Context, 1)
+				return err
+			},
+			expectedError: types.ErrEarliestAvailableNonceStillNotInitialized,
+		},
+		{
+			name: "check earliest available attestation nonce before getting latest data commitment",
+			requestFunc: func() error {
+				_, err := k.GetLatestDataCommitment(input.Context)
+				return err
+			},
+			expectedError: types.ErrEarliestAvailableNonceStillNotInitialized,
+		},
+	}
```

Comment:
do we have these, and the other tests besides TestCheckLatestAttestationNonce, just to ensure that we are calling `CheckLatestAttestationNonce` in those functions?

Reply:
If those functions were not calling the check method, that error will not be returned. The main reason we're checking for that specific error is to see if the check is called.

Is this what you meant? Or im missing your question

Reply:
yeah that makes sense. I was just making sure that I understood the purpose of the tests.